File: frontend/src/types/bowler.ts
// These names need to be just like they are in the incoming json file
export type Team = {
  teamID: number;
  teamName: string;
  captainID: number;
};

export type bowler = {
  bowlerID: number;
  bowlerFirstName: string;
  bowlerMiddleInit: string;
  bowlerLastName: string;
  bowlerAddress: string;
  bowlerCity: string;
  bowlerState: string;
  bowlerZip: string;
  bowlerPhoneNumber: string;
  teamName: Team; // This is where we are passing the team which includes the teamID, teamName, and captainID.
};

File: backend/Alder_Mission10/Alder_Mission10/Data/Bowler.cs
﻿using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Alder_Mission10.Data
{
    public class Bowler
    {
        [Key]
        public int BowlerID { get; set; }
        public string? BowlerFirstName { get; set; }
        public string? BowlerMiddleInit { get; set; }    
        public string? BowlerLastName { get; set; }  
        public string? BowlerAddress { get; set; }
        public string? BowlerCity { get; set; }
        public string? BowlerState { get; set; }
        public string? BowlerZip { get; set; }
        public string? BowlerPhoneNumber { get; set; }

        [ForeignKey("TeamID")]
        public int? TeamID { get; set; }
        public Team? TeamName { get; set; }
    }
}
